### PR TITLE
feat: added 'Custom' param that allows async getter and custom validator

### PR DESCRIPTION
**Pull request by MacMcIrish**

### CODE REVIEW

File: src/param.js [hunk 1]
```diff
@@ -418,3 +418,25 @@ class NumberParam extends Json {
   }
 }
 module.exports.Number = (...args) => new NumberParam(...args);
+
+class Custom extends Param {
```
Comment (MacMcIrish) [inline src/param.js]: @simlu `Custom` is just a placeholder, I'm wondering if you already had a name in mind for this type of param?

Comment (simlu) [reply src/param.js]: Maybe `Async`?

Comment (simlu) [reply src/param.js]: Or should this just be a "remap" parameter (optional) in Param?

File: test/handler.js [hunk 2]
```diff
@@ -16,6 +17,17 @@ module.exports.proxy = api.wrap('GET proxy/{proxy+}', [
   api.Str('proxy+', 'path')
 ], process.env.RATE_LIMIT, ({ proxy }) => api.JsonResponse({ path: proxy }));
 
+const externalRequest = value => (value === undefined
```
Comment (simlu) [inline test/handler.js]: why do we need to handle undefined?

File: test/handler.js [hunk 3]
```diff
@@ -1,3 +1,4 @@
+const https = require('https');
```
Comment (simlu) [inline test/handler.js]: use dev dep ext library

File: src/param.js [hunk 4]
```diff
@@ -14,7 +14,7 @@ const positionMapping = {
 };
 
 class Param {
-  constructor(name, position = 'query', required = true, { nullable = false } = {}) {
+  constructor(name, position = 'query', required = true, { nullable = false, getter = null } = {}) {
```
Comment (simlu) [inline src/param.js]: document!

File: README.md [hunk 5]
```diff
@@ -59,6 +59,55 @@ The `defaultHeaders` are returned with every request that isn't an unexpected cr
...
+], ({ name }) => {
+  console.log(name); // null
+});
+```
+
+#### options.getter
+
+_*Note: only recommended for advanced use cases.*_<br>
+Optionally asynchronous custom "getting" of variables.<br>
+Getter function takes raw input from event, IE a `query` parameter will always pass String values into the `getter` function.<br>
```
Comment (simlu) [inline README.md]: Is that actually true?

**Pull request merged**